### ISSUE TITLE
Generic.CodeAnalysis.UnusedFunctionParameter.Found false positive with func_get_args()

### ISSUE THREAD

**Issue by LavaIsCool** [squizlabs/PHP_CodeSniffer#988]
**Example:**

```
public function doSomething($param1 = null, $param2 = false) {
    $func_args = func_get_args();
    return call_user_func_array(array('someClass', 'someFunction'), $func_args);
}
```

Generic.CodeAnalysis.UnusedFunctionParameter.Found reports the following errors:

```
The method parameter $param1 is never used
The method parameter $param2 is never used
```

**Comment by aik099** [comment]
Technically it's not used. We can't know that easily (without tracing usage of variable, where `func_get_args()` were assigned) that parameters are actually used.

I recommend using PHPMD to detect unused stuff instead.

**Comment by LavaIsCool** [comment]
Perhaps the error message in this case could be changed to something like 'The method parameter $param might not be used' or something? Seems like that would be more appropriate in this scenario.

**Comment by glen-84** [comment]
I just ran into this with the `compact` function.

```php
public function viewAction($id, $username)
{
    $this->view->setParams(compact('id', 'username'));
}
```

This is probably easier to handle than the above example, but I'll probably just avoid the usage for now.

**Comment by u01jmg3** [comment]
As per the example given by @glen-84, I am getting an unused variable warning when `compact()` is used. Should the sniff be enhanced to recognise compact usage?

**Comment by gsherwood** [comment]
Closing as I don't think this is a core part of what PHPCS does. The sniffs that check for things other than code consistency should probably be deprecated in favour of other tools that focus on this kind of analysis.
